### ISSUE TITLE
Scrollytelling: SALURBAL City Selection

### ISSUE THREAD

**Issue by ran-codes** [Drexel-UHC/salurbal-city-selection-scrolly#1]
### Pending Issues

- [ ] Drexel-UHC/salurbal-city-selection-scrolly-spanish#1 
- [ ] Drexel-UHC/salurbal-city-selection-scrolly#3 

### Action Items

- [x] Sent out for review to DMC and Executive Team. Deadline for comments is 6/21
- [x] Feedback compiled: [Scrolly - Salurbal City Selection-6 11Aug2023.docx](https://github.com/Drexel-UHC/salurbal-project-dashboard/files/12380618/Scrolly.-.Salurbal.City.Selection-6.11Aug2023.docx)
- [x] @ran-codes to update scrolly
   - [x] ETL required boundaries
      - [x] All city centroids
      - [x] SP official city boundaries (L2 10224810)
      - [x] SP L1AD
      - [x] SP L1UX
      - [x] SP L2
      - [x] SP L2_5
      - [x] SP L3 
   - [x] add centroids for SP (step 1)
   - [x] add boundaries for Municipio (step 2)
   - [x] add L1UX
   - [x] adapt MapLayer to update paint based on action-custom-paint 
- [x] @usamabilal to review

### Discussed in https://github.com/Drexel-UHC/salurbal-project-dashboard/discussions/101

<div type='discussions-op-text'>

<sup>Originally posted by **usamabilal** February  3, 2022</sup>
Following OLS idea of putting details on city selection, I think a great idea would be to create a "visual essay" summarizing it.
Examples provided by @rl627 in another thread:
From google's ML group: https://pair.withgoogle.com/explorables/hidden-bias/
from a 'visual essay' news outlet: https://pudding.cool/2017/01/shape-of-slavery/
undergraduate major story: https://cuthchow.github.io/college-majors-visualisation/

My idea for this would be:

1. How did we select cities? Brief text explaining AUE and citypopulation.de and selecting cities based on >100k on 2010. 
2. Zoom into a city "core" (e.g. CABA in Buenos Aires). Show satellite imagery. Something about how this is the city core of buenos aires with 3 million people etc. The city itself is just those 3 million people.
3. Zoom out to the entire urban extent of the metropolitan area. Something about how this has 16 million people, and represents a continuous urban extent.
4. Overlay L2s, and "draw" (manually) the urban extent. Say something about how we selected local administrative units that overlap with this. 
5. Shade somehow those that end up being part of buenos aires L1AD. Say these are the LAUs that are part of the L1AD. 
6. "Merge" all L2s into the L1AD, and say "This is what we call Level 1, the city". Can copy text from brief on what L1 is.
7. "Zoom in again" and show L2s again. Say these we call L2s, subcities. Copy text from brief
8. "Zoom in again" and show L3s. Say these we call L3s, neighborhoods. Copy text from brief
9. Zoom out to the entire region, and just have a dot per city, and say we repeated this process for the 371 cities in x, y and z.

(all feedback is welcome!)

useful resources:

- data brief (lots of useful text): https://drexel.edu/~/media/Files/lac/Data/Data%20Brief_ENG_FINAL_web.ashx?la=en. this is also translated into Spanish and Portuguese in case we wanted the essay to be translated too
- the infamous city selection document: https://drive.google.com/file/d/1LatY_LP5VlpJWZL8VjdiBWIJK_iqPNr1/view
</div>

### Resources dump

- React-land scrolling implementation: https://github.com/jsonkao/react-scrollama
- D3 over Google map's satellite background: https://bl.ocks.org/mbostock/899711
- D3 animated transitions: https://observablehq.com/@mprasse/week-12-interaction-and-animation-d3-transitions-behavior

### Draft v1

- [x] Sketch panels in R and check with @usamabilal 
- [x] Set page ('city-selection') for viz essay and add to appBar
- [x] Set up scrollarama layout (https://github.com/jsonkao/react-scrollama)
- [x] Set up geojson in datastore
- [x] Vanilla javascript scrollytelling: get scrolling events + state management + linkages to map layers done.
- [x] Create a Scrollarama side-sticky template
- [x] Translate JS implementations content to scrollarama template
- [x] Leaflet
    - [x] update boundary based on scrollState
    - [x] update zoom based on scrollState
    - [x] update boundary layer styling based on scrollState
    - [x] Remove zoom control and attributions.  
 - [x] first polish of UI ( first pass at layout/colors/etc... )
    - [x] copy text from brief into scrolly
    - [x] OSM tile white lines (https://github.com/Leaflet/Leaflet/issues/3575#issuecomment-688644225)
    - [x] zlow down zoom (https://stackoverflow.com/questions/38653793/how-to-zoom-gradually-in-leaflet-javascript) 
 - [x] BUild and deploy
     - [x] react-leaflet is causing build bugs. (#108)
     - [x] Deploy as staging link
 - [x] review with @usamabilal @sjmelly @kam642 @andreabolinaga 

### Draft v2

- [x] page transition bug (city-selection) -> city_profile
- [x] Client side debounce bug when scrolling too fast
- [x] Add L3 geojson
- [x] Make CitySelectionScrolly robust to multiple city data
    - [ ] Import LAU names
    - [ ] configure data store 
    - [ ] create dynamic pages for each city (deal with staticProps and getStaticPaths)
    - [ ] make the CABA slide specific only to BA
    - [ ] Add buttons to CIty profile pages
- [x] centroids map make interactive
    - [ ] Add tooltips
    - [ ] Clicking centroid will push to city specific scrolly.

**Comment by ran-codes** [comment]
Thanks for the feedback @kindvik!!

Updates (remember to hard refresh cntrl+R) :
- ADR text updates: https://drexel-uhc.github.io/salurbal-city-selection-scrolly/
- Spanish version generated: https://drexel-uhc.github.io/salurbal-city-selection-scrolly-spanish/

 @usamabilal @kindvik how does the Spanish version look?

**Comment by ran-codes** [comment]
new link for the spanish version. please review here: https://mellow-blancmange-4aad09.netlify.app/?review=true

@kindvik there is a new tool for giving feedback we are trying. can you try it out (see slnip below)? (you have to create an account with gmail then sign in to leave comments). 

![image](https://github.com/Drexel-UHC/salurbal-city-selection-scrolly/assets/45013044/bb66cda2-6a4a-4a5f-b59c-68b7d82eb22b)

**Comment by kindvik** [comment]
Hi Ran,
This looks great - thank you! I tried out the new tool and found it intuitive; is this something we could utilize across the data portal?
I left a number of comments on Spanish version; Usama of course feel free to veto. Thanks again for moving this forward!
Katy

**Comment by ran-codes** [comment]
@kindvik thanks for the comments.  I will make the updates  .

### Re: reporting tool. 

Glad you like it. I imagine something like this will save us a lot of time in regards to future web projects. We have implemented a prototype of it for the data portal already. here is the stable link for the review version of the data portal: https://wonderful-moss-0f45f310f-393.eastus2.azurestaticapps.net/?review=true

**Comment by usamabilal** [comment]
Thanks @kindvik ! i only "vetoed" two for grammar reasons.

**Comment by kindvik** [comment]
And @ran-codes -- absolutely!! This will save a ton of time for all, I think!!

**Comment by ran-codes** [comment]
How is this issue still open... ? haha

Thank you to everyone involved. ❤️ This was one of my favorite projects to work on in my time with UHC and SALURBAL. It's incredibly difficult to communicate complexity, but I think we did a great job pulling it off. 

BTW kudos to @usamabilal ... who had the original prompt for this -  [see this discussion](https://github.com/Drexel-UHC/salurbal-project-dashboard/discussions/101)

Closing 😄
